arangod/IResearch/IResearchLink.cpp

```diff
@@ -1516,12 +1753,26 @@ Result IResearchLink::initDataStore(
   options.meta_payload_provider = [this](uint64_t tick, irs::bstring& out) {
     // call from commit under lock _commitMutex (_dataStore._writer->commit())
     // update last tick
-    _lastCommittedTick = std::max(_lastCommittedTick, TRI_voc_tick_t{tick});
+    auto lastCommittedTick = _commitStageOne ? &_lastCommittedTickStageOne
+                                             : &_lastCommittedTickStageTwo;
+    *lastCommittedTick = std::max(*lastCommittedTick, TRI_voc_tick_t{tick});
```

Comment:
mmh, we are `hton64`ing the value read here _later_, but not here for determining the maximum. I think to be safe, we need to hton64 it here to, when comparing it to `tick`.

Reply:
Please ignore my comment. It is not relevant.